### ISSUE TITLE
PROBLEM: Can't reach paper results using exact same configuration and dataset

### ISSUE THREAD

**Issue by jinchengluo** [lucadellalib/focalcodec#13]
Hello @lucadellalib ,

I have been working with FocalCodec for some time right now but I ran into some issues concerning the training of the model.
To run the model trainings, I used the provided speechbrain recipe (https://github.com/speechbrain/speechbrain/tree/develop/recipes/LibriTTS/focalcodec) and kept all the same hyperparameters and training strategies which should be the same as stated in the paper.

However, after a really long time of training the model on LibriTTS train-clean-100 split, I can't reach the same results stated in the paper.

When looking at the Tensorboard logs you provided, it looks like the discriminator quickly improves for around 500k steps of gradient update for the discriminator loss and after that it increases back to finally converge, from 750k steps, around the value of 9. The total generator loss follows the opposite tendency before 750k steps and also converges afterward.

In my case, when looking at the sub losses, especially the adversarial loss and the feature matching loss, it slowly and constently increases whereas the discriminator loss decreases in the same manner. It never goes back up.

I trained for 1M+ steps. Hence, I am very doubtful of my results as it should be the same as yours.

I would be very happy if you could clarify why I encounter this type of problem using the same model and training configuration.

My losses:

<img width="1525" height="1017" alt="Image" src="https://github.com/user-attachments/assets/d995001d-7a5e-4a5f-aea4-ac62342f7012" />

Yours:

<img width="1527" height="898" alt="Image" src="https://github.com/user-attachments/assets/1819a06b-962b-49a9-a6a0-c15d48dbbb28" />

Thank you in advance !
Best,
